lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -1523,6 +1523,11 @@ namespace {
       if (!isSendableClosure(closure, /*forActorIsolation=*/false))
         return;
 
+      // Skip this warning for implicit closures such as async let
+      // implicit generated autoclosures.
+      if (closure->isImplicit())
+        return;
+
```

Comment:
I don't think we can skip the Sendable checking on captures like this. The autoclosure for an async-let needs to be a `@Sendable` closure, since it's going to be run in another task. Skipping the checking for non-sendable captures puts a hole in the concept of Sendable, since these warnings are going to turn into errors in Swift 6.

Reply:
It does make sense, I've added test cases to exhaust Sendable and non-sendable types capture in implicit auto closure and one TODO to improve the diagnostic wording, because I think from the user perspective the current warning today in the implicit autoclosure case can be a little bit confusing.